src/detect-dns-query.c

```diff
@@ -1256,13 +929,9 @@ static void DetectDnsQueryRegisterTests(void)
     UtRegisterTest("DetectDnsQueryTest01", DetectDnsQueryTest01);
     UtRegisterTest("DetectDnsQueryTest02", DetectDnsQueryTest02);
     UtRegisterTest("DetectDnsQueryTest03 -- tcp", DetectDnsQueryTest03);
-    UtRegisterTest("DetectDnsQueryTest04 -- tcp splicing",
-                   DetectDnsQueryTest04);
```

Comment:
Why remove these tests ?

Reply:
A replacement for `DetectBsizeSigTest01` was added to the Rust test set. These tests require TCP buffering which isn't possible when directly invoking the application layer parsing.

Reply:
> These tests require TCP buffering which isn't possible when directly invoking the application layer parsing.

It is possible as I did recently when translating the SSH parser to Rust and using ApplayerResult::Incomplete
You can see in `DetectSshVersionTestDetect01` cf https://github.com/OISF/suricata/blob/master/src/detect-ssh-proto-version.c#L339

And I think it is particularly useful to keep these tests when doing such a change ;-)

Reply:
From the DNS perspective I think its probably OK to not test this as a unit test. We should have tests that that check the correct response is returned when the data is incomplete.

While there should be some larger tests that test that the buffering +  protocol parsing does work, it probably does not need to be implemented for every protocol.

What might be more intersting is suricata-verify tests with pcaps that cause the parsers to be called with incomplete data.

Reply:
Related: https://redmine.openinfosecfoundation.org/issues/3725

Reply:
I agree that they can be replaced by suricata-verify tests.

I wish I had known this ticket while translating the SSH tests ;-)